lightning/src/routing/router.rs

```diff
@@ -71,9 +71,10 @@ impl< G: Deref<Target = NetworkGraph<L>>, L: Deref, S: Deref,  SP: Sized, Sc: Sc
 			*locked_random_seed_bytes = Sha256::hash(&*locked_random_seed_bytes).into_inner();
 			*locked_random_seed_bytes
 		};
+		let scorer: Box<dyn DerefMut<Target = Sc>> = Box::new(self.scorer.lock());
```

Comment:
Still working on this..
this line still has the error
```
the size for values of type `dyn DerefMut<Target = Sc>` cannot be known at compilation time
the trait `Sized` is not implemented for `dyn DerefMut<Target = Sc>`
```

Reply:
Ok fwiw, this is because `DerefMut` isnt object safe and therefore `dyn DerefMut<...>` causes the compiler to not be able to deal with it. 

Thinking out loud here .. The compiler thinks this is a silly idea, which makes some sense because what this code kinda means is "i want to pass you some box with unsized ref which later on, I want back a mutable reference of the concrete class sitting in the box". which sounds silly, why didnt you:
1/ pass me the concrete owned/mutable reference to begin with
2/ only call immutable methods on the trait object going forward, since usage of the trait object already comes with some type erasure.

its likely that we can do 1 with some more generics magic, which is what @TheBlueMatt is suggesting.